src/infoPanel/infoPanel.c

```diff
@@ -88,10 +148,56 @@ int main(int argc, char *argv[])
 	while (!quit && wait_confirm) {
 		while (SDL_PollEvent(&event)) {
 			if (event.type == SDL_KEYDOWN)
-				quit = true;
+			{
+				bool navigation_pressed = false;
+				bool navigating_forward = true;
+				switch(event.key.keysym.sym) {
+				case SW_BTN_A:
+					navigation_pressed = true;
+					navigating_forward = true;
+					break;
+				case SW_BTN_B:
+					navigation_pressed = true;
+					navigating_forward = false;
+					break;
+				default:
+					break;
+				}
+				if (!navigation_pressed) 
+				{
+					continue;
+				}
+				if ((navigating_forward && image_index == images_count-1) // exit after last image
+					|| (!navigating_forward && image_index == 0)) // or when navigating backwards from the first image
+				{
+					quit = true;
+				}
+				else
+				{
+					if (navigating_forward)
+					{
+						image_index++;
+					}
+					else
+					{
+						image_index--;
+					}
+					drawImage(images_paths[image_index], screen);
```

Comment:
It seems quite inefficient to load the images every time they're drawn? It's the navigation slowed by this. You just need to store the surface pointers in an array, and remember to free each one at exit.

Reply:
fixed